src/ControlSystem/WriteData.hpp

```diff
@@ -0,0 +1,174 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "ControlSystem/Component.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/QuaternionFunctionOfTime.hpp"
+#include "IO/Observer/Helpers.hpp"
+#include "IO/Observer/ObservationId.hpp"
+#include "IO/Observer/ReductionActions.hpp"
+#include "IO/Observer/TypeOfObservation.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/Info.hpp"
+#include "Parallel/Invoke.hpp"
+#include "Parallel/Reduction.hpp"
+#include "Utilities/Functional.hpp"
+#include "Utilities/MakeString.hpp"
+
+namespace control_system {
+namespace detail {
+struct WriterHelper {
+  // Use funcl::AssertEqual for all because we aren't actually "reducing"
+  // anything. These are just one time measurements.
+  using ReductionData = Parallel::ReductionData<
+      // Time
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      // Lambda, dtLambda, d2tLambda
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      // Q, dtQ, control signal
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>>;
+
+  using observed_reduction_data_tags =
+      observers::make_reduction_data_tags<tmpl::list<ReductionData>>;
+
+  const static inline std::vector<std::string> legend{
+      "Time", "Lambda", "dtLambda", "d2tLambda", "Q", "dtQ", "Control Signal"};
```

Comment:
bigger question: is `Q` just the deviation between what we want Lambda to be and what it is? If so, I would strongly prefer we call it `Deviation` instead of `Q`. Single-letter names for things are always difficult to understand and rarely very descriptive...

Reply:
`Q` is the control error so in general, yes, `Q` is the deviation between what we want `Lambda` to be and what it is. However, it doesn't always take the form of `expected - actual` (several control systems have more complicated formulas for `Q`). The reason I chose `Q` is because [this paper](https://arxiv.org/abs/1211.6079) and [this paper](https://arxiv.org/abs/1304.3067) define most, if not all, the control systems in SpEC and the ones we will use in SpECTRE. We've always used a capital `Q` to denote the control error. I think changing it to something else here would make it confusing for anybody trying to understand where these things came from. For that reason, in this case, I'd prefer to leave the single-letter name.

Reply:
I feel rather strongly about not using single letter names, especially ones that aren't easy to understand. I've always found use Q confusing in SpEC because whenever I've had to diagnose control system problems it comes up and nobody knows what it means other than Mark. The "It's always been done like this" argument is not a good one to keep doing something. It sounds like the correct name for this is "ControlError". If we only used notation from papers then I think the entire code would be impenetrable (look at SpEC hydro...).

Reply:
I like `ControlError` and `dtControlError`.